### DISCUSSION TITLE
API for sponsorship

### DISCUSSION THREAD

**Discussion by GregSlazinski** [community/community#24649] [Apps, API and Webhooks]
<p>Is there an API to access sponsorship information?</p>
<p>What I’d like to have:</p>
<p>connect via PHP to GitHub</p>
<ol>
<li>check the total amount of money earned per month</li>
</ol>
<p>as displayed here:</p>
<p><a href="https://github.com/sponsors/GregSlazinski/dashboard" rel="nofollow noopener">https://github.com/sponsors/GregSlazinski/dashboard</a></p>
<ol start="2">
<li>obtain the list of sponsors (their email address) and their sponsor tier ($ per month)</li>
</ol>
<p>having known their email address and tier, I can automatically grant badges/ranks/colors on my own forum for my supporters.</p>
<p>Thanks,</p>
<p>Greg</p>
<p><a href="https://github.com/Esenthel/EsenthelEngine/" rel="nofollow noopener">https://github.com/Esenthel/EsenthelEngine/</a></p>

**Comment by AndreaGriffiths11** [comment]
<p>Hi <a class="mention" href="/gregslazinski">@gregslazinski</a>,</p>
<p>Thank you for being here! Per @<strong><a href="https://github.com/devonzuegel" rel="nofollow noopener">devonzuegel:</a></strong></p>
<p><strong>"</strong> We’re currently building an Activity Timeline that shows a bunch of different types of events, including transactions. We’re also planning to build a transaction CSV download that you can export to do this sort of analysis on. We’re projecting that both of these will be out in the next month or two. 😊</p>
<p>In the meantime, you could look at (a) what tier a sponsor was on and (b) look up when you received the celebration email saying that they’ve started sponsor you as a fairly good approximation of how much they’ve sponsored you in total."</p>
<p>To read more about “Viewing your sponsors and sponsorships” and “Contacting your sponsors”:</p>
<ul>
<li><a href="https://help.github.com/en/articles/viewing-your-sponsors-and-sponsorships" rel="nofollow noopener">https://help.github.com/en/articles/viewing-your-sponsors-and-sponsorships</a></li>
<li><a href="https://help.github.com/en/articles/contacting-your-sponsors" rel="nofollow noopener">https://help.github.com/en/articles/contacting-your-sponsors</a></li>
</ul>
<p>I hope this helps!</p>

**Comment by devonzuegel** [comment, marked as answer]
<p>You can learn about the Sponsors API and see public sponsor-related schema info by going to <a href="https://developer.github.com/v4/explorer" rel="nofollow noopener">developer.github.com/v4/explorer</a> and searching <code>"sponsor"</code> in the search bar to the right, halfway down the page:</p>
<p><img src="https://camo.githubusercontent.com/851a78138d8f106915de0fcdad1672b26f6b6dfd/687474703a2f2f672e7265636f726469742e636f2f6b3545687362486854792e676966" alt=""></p>

**Comment by tinspin** [comment]
<p>My post got removed? How would you write GraphQL to see if a particular user has sponsored you? I don’t want to paginate through all the sponsors, specially when the rateLimit is 5000 requests per hour!?</p>

**Comment by erikaheidi** [comment]
<p>+1 for a REST API endpoint where we can check if a user sponsors the authenticated user, an at what tier. This would enable us to do a lot more!</p>

**Comment by ArsalaBangash** [comment]
<p>Is there any progress on this?</p>

**Comment by giuseppeg** [comment]
<p><a class="mention" href="/devonzuegel">@devonzuegel</a> it seems that the API returns <code>null</code> when trying to query for the tiers eg.</p>
<pre><code class="lang-auto">{
  viewer {
    sponsorshipsAsSponsor(first: 10) {
      nodes {
        tier {
          name
          description
          id
        }
      }
    }
  }
}
</code></pre>
<p>Another dev reported the same issue on StackOverflow <a href="https://stackoverflow.com/questions/60281942/github-graphql-getting-sponsor-tier-information-for-the-sponsors-of-a-user" rel="nofollow noopener">https://stackoverflow.com/questions/60281942/github-graphql-getting-sponsor-tier-information-for-the-sponsors-of-a-user</a></p>
<p>I’d appreciate if you could help us figure this out :slight_smile: Thank you in advance for your time!</p>

**Comment by andig** [comment]
<p>So, as of to date, it is still impossible to determine which users have sponsored me via api?</p>

**Comment by EnviralDesign** [comment]
<p>Also interested in having the ability to access this information in a cleaner more reliable way than web scraping!</p>

**Comment by adamghill** [comment]
<p>I was also confused that the sponorship tier always seemed to be null no matter what permissions I requested, however it does seem to work if a “personal access token” is used as authorization for GraphQL. I’m not sure why it doesn’t work with the regular OAuth token that gets generated, though.</p>

**Comment by liskin** [comment]
<p>Can we please have a way to distinguish private vs. public sponsors through the GraphQL API? I’m trying to <a href="https://github.com/xmonad/xmonad-web/pull/39" rel="noopener nofollow ugc">automate listing of sponsors on our webpage</a> but the API gives me <strong>all</strong> sponsors, not just public ones, meaning I need to double check this every time I run the script. :frowning:</p>

**Comment by jrochkind** [comment]
Hi, I am very interested in being able to get a list of *public* sponsors from API (with tiers), to include them on a web page thanking them. 

This seems like a core use case for github sponsors, right? 

I am new to GraphQL so was having trouble figuring out how to do this... but it seems that it actually may not be currently possible?  @liskin do you have an update from two years ago, does the situation remain the same?   

It would take me a while to figure out how to do this with GraphQL even if it is possible (athank you for linking to your code, to give me a head start), but if it is not in fact possible I'll save my time! 

Is there any place we can formally register a request for this feature?

**Reply by liskin** [reply]
Since writing that post, the GraphQL API improved and it is actually now possible to do this. [sponsorshipsAsMaintainer](https://docs.github.com/en/graphql/reference/interfaces#:~:text=sponsorshipsAsMaintainer) lets you say `includePrivate: false`. The [current version of the script](https://github.com/xmonad/xmonad-web/blob/0407056255cc253da74a586ef80c4ed16c51d75e/_sponsors.sh) that refreshes named sponsors at https://xmonad.org/community.html uses this new API.

(Fun fact: before that API appeared, we had a workaround that used two different GitHub API tokens. One had admin access to the xmonad org and could see all sponsors and their sponsorship levels, and the other was just the token you get from GitHub Actions that could only see public sponsors and no details. That was enough to piece it all together. But now you can just query everything easily using the admin token and the GraphQL API.)

**Reply by jrochkind** [reply]
thanks for the update!  And the info on that workaround too!
